### PR TITLE
Make an auto labeler action 

### LINKED ISSUE #118: Add an autolabeler
Add this GitHub Action autolabeler to the repo:
https://github.com/actions/labeler

- for the first pass, each directory should have a label with the directory name, downcased
- the readme and contributing docs should have a _documentation_ label

### PR DESCRIPTION
**Pull request by weshaan**
# Pull Request

## Description
I am adding a github action that automatically labels pull requests.

## Motivation
solves issue #132 

## Testing Procedures
make changes in any file/add a new file in any folder and create a pull request for it. the labeler will run a check and add necessary labels. 

## Checklist
- [yes ] Code adheres to project coding standards.
- [yes ] Tests cover the added or modified code.
- [yes ] Documentation has been updated, if necessary.
- [yes ] All new and existing tests pass successfully.
- [yes ] The pull request is up to date with the latest changes from the main branch.

## Reviewer Guidelines
i labeled "assets/readme" as documentation cause it affects the readme. Pls let me know your thoughts on it. 

## Screenshots or Visuals
image

## Additional Information
Thank you. 

Fixes #132

### CODE REVIEW

**Comment by weshaan** [comment]
Hey! can someone please update me about this?

**Comment by JenBauer** [comment]
Hi @weshaan, there was no reviewer added to this PR, so I added one.  Also, can you test this action in your forked repository to show that the labeler is working as intended?  I'm not seeing that any actions have run on your side, but that would be the easiest way to confirm that this is working as expected.

**Comment by weshaan** [comment]
hey, I was unsure initally if I could run actions on a forked repo sorry 😅 . I made some fixes and it works! 
image
I will check some different cases and once all of them show the expected results I'll propose a new pull request 👍

**Comment by weshaan** [comment]
Please review my new pr. Thank you!

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
+2023_hacktoberfest:
+  - 2023_Hacktoberfest/**
+
+eventvideos:
```
Comment (klmansel) [inline .github/labeler.yml:5]: This one can go under the `documentation` label

**Comment by weshaan** [comment]
Thank you for answering. Should I push this?
image

**Comment by klmansel** [comment]
That works, thanks! You can just update the original branch and PR. You don't have to open a new PR for each edit.

**Comment by weshaan** [comment]
okay, I think this PR is ready to be merged then 👍

**Comment by pksuperduper** [comment]
This is cool!  Since not everyone has admin access to add labels.  I am sort of new to yaml files.  How can I try this out in a local environment?

**Comment by weshaan** [comment]
> This is cool -- since not everyone has admin access to add labels. I am sort of new to yaml files. How can I try this out in a local environment?

Hey, I wrote a blog about what I understood doing this if you wanna check that out :)
[Blog page](https://weshaan.hashnode.dev/using-github-actions-to-add-labels-automatically)

**Comment by weshaan** [comment]
@klmansel Hey, Thank you again for reviewing my PR. are we ready to marge it?

**Comment by JenBauer** [comment]
Hi @klmansel, has this cleared review and is set now to merge?

**Review by klmansel — APPROVED** [review]
thank you @weshaan!

**Pull request merged**